pool/sender_pool_test.go

```diff
@@ -0,0 +1,106 @@
+package pool_test
```

Comment:
Tests for TCP senders are missing.

Reply:
Do we really need a dedicated package for such simple thing as pool? To me, it's not worth it and I'd keep everything in a single `questdb` package. The client's code base is small.

Reply:
I think that it's a clean separation of concerns because pool works on the `LineSender`'s interface level. And this way, there's no confusion between which primitive to use, since in some cases, you may not need the pool and just want to reach for a single sender.

Finally, I think that this subpackage abides by the [go best practices of package naming](https://go.dev/doc/effective_go#package-names). So a user can create a pool semantically using the simpler `pool.FromConf` instead of `questdb.PoolFromConf` or `v3.PoolFromConf`.

But if you feel strongly about the single package, I'm happy to move everything back into the `v3` package

Reply:
I don't see anything wrong with `qdb.PoolFromConf`. Sub-packages should be introduced consistently and having a dedicated package just for the pool may be weird since everything else is available under a single package. But it's up to you - if you think introducing a package is a good idea, let's do it.

Reply:
> Sub-packages should be introduced consistently 

That's fair, I'll move everything into the main package. We can think about splitting up subpackages in a future major version